platform/linux-generic/odp_ipsec.c

```diff
@@ -1917,6 +1917,57 @@ int odp_ipsec_out_inline(const odp_packet_t pkt_in[], int num_in,
 	return in_pkt;
 }
 
+int odp_ipsec_sa_info(odp_ipsec_sa_t sa, odp_ipsec_sa_info_t  *sa_info)
+{
+	ipsec_sa_t *ipsec_sa;
+	odp_ipsec_dir_t dir;
+
+	ipsec_sa = _odp_ipsec_sa_entry_from_hdl(sa);
+
+	ODP_ASSERT(ipsec_sa != NULL);
+	ODP_ASSERT(sa_info != NULL);
+
+	memset(sa_info, 0, sizeof(odp_ipsec_sa_info_t));
+
+	dir = ipsec_sa->param.dir;
+
+	sa_info->param = (odp_ipsec_sa_param_t)ipsec_sa->param;
+
+	if (dir == ODP_IPSEC_DIR_OUTBOUND) {
+		sa_info->outbound.seq_num =
+			(uint64_t)odp_atomic_load_u64(&ipsec_sa->hot.out.seq);
+
+		if (ipsec_sa->param.mode == ODP_IPSEC_MODE_TUNNEL) {
+			uint8_t *src, *dst;
+
+			if (ipsec_sa->param.outbound.tunnel.type ==
+			    ODP_IPSEC_TUNNEL_IPV4) {
+				src = sa_info->outbound.tunnel.ipv4.src_addr;
+				dst = sa_info->outbound.tunnel.ipv4.dst_addr;
+				memcpy(src,
+				       &ipsec_sa->out.tun_ipv4.src_ip,
+				       _ODP_IPV4ADDR_LEN);
+				memcpy(dst,
+				       &ipsec_sa->out.tun_ipv4.dst_ip,
+				       _ODP_IPV4ADDR_LEN);
+			} else {
+				src = sa_info->outbound.tunnel.ipv6.src_addr;
+				dst = sa_info->outbound.tunnel.ipv6.dst_addr;
+				memcpy(src,
+				       &ipsec_sa->out.tun_ipv6.src_ip,
+				       _ODP_IPV6ADDR_LEN);
+				memcpy(dst,
+				       &ipsec_sa->out.tun_ipv6.dst_ip,
+				       _ODP_IPV6ADDR_LEN);
+			}
+		}
+	} else {
+		ODP_ERR("Feature to be added\n");
```

Comment:
I think this needs to be implemented before this can be merged.

Reply:
This is still needed.

Reply:
In v10, implemented inbound specific param retrieval for the SA info API.